DoublyLinkedList/doubly_linked_list.h

```diff
@@ -0,0 +1,287 @@
+#include <iostream>
+using namespace std;
+// Node Creation
+class node
+{
+public:
+    int data;
+    node *next;
+    node *prev;
+};
+
+class doubly_linked_list : private node
+{ //class
+    node *head=nullptr;
+    node *tail=nullptr;
+    int length = 0;
+
+public:
+    //for creating new node
+    node* create_node(node *x, int value, node *y);
+
+    // insertion of value at front & back
+    void push_front(int value);
+    void push_back(int value);
+
+    //for inserting at location
+    void insert(int value, int location);
+
+    //for removing values at front and back
+    int remove_front();
+    int remove_back();
+
+    //for removing at a location and from a range
+    void remove(int location);
+    void remove(int start, int end);
+
+    //for finding the size
+    size_t size();
+
+    //check whether empty or not
+    bool is_empty();
+
+    //Printing of values
+    void print_values();
+
+    //to clear all the elements in the list
+    void clear();
+
+};
+//for  creating new node
+node* doubly_linked_list::create_node(node *x, int value, node *y){
+    node *new_node = new node();
```

Comment:
pl move this as a constructor to node. 

node(int value, node * prev, node *next) 

we dont need a create node function.